hail_search/queries/both_sample_types_mixin.py

```diff
@@ -0,0 +1,141 @@
+            )
+        )
+        return ht.transmute_globals(family_guids=ht.family_guids.extend(ht.family_guids_1))
+
+
+    def _filter_inheritance_both_sample_types(
+        self, ht, inheritance_filter, sorted_wes_family_sample_data, sorted_wgs_family_sample_data
+    ):
+        # At least one family member must have non-ref gt
+        any_valid_entry, is_any_affected = self._get_any_family_member_gt_has_alt_filter()
+        ht = ht.annotate(
+            wes_passes=ht.wes_family_entries.map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            ),
+            wgs_passes=ht.wgs_family_entries.map(
+                lambda entries: hl.or_missing(entries.any(any_valid_entry), entries)
+            )
+        )
+
+        comp_het_ht = None
+        if self._has_comp_het_search:
+            comp_het_ht = self._annotate_families_inheritance(
+                ht, COMPOUND_HET, inheritance_filter, sorted_wes_family_sample_data,
+                sample_type='WES',
+            )
+            comp_het_ht = self._annotate_families_inheritance(
+                comp_het_ht, COMPOUND_HET, inheritance_filter, sorted_wgs_family_sample_data,
+                sample_type='WGS',
+            )
+            comp_het_ht = self._post_process_inheritance_both_sample_types(comp_het_ht)
+            comp_het_ht = comp_het_ht.filter(
+                comp_het_ht.wes_passes.any(hl.is_defined) | comp_het_ht.wgs_passes.any(hl.is_defined)
+            ).select_globals('family_guids', 'family_guids_1')
+
+        if is_any_affected or not (inheritance_filter or self._inheritance_mode):
+            # No sample-specific inheritance filtering needed
+            sorted_wes_family_sample_data = sorted_wgs_family_sample_data = []
+
+        if self._inheritance_mode == COMPOUND_HET:
+            ht = None
+        else:
+            ht = self._annotate_families_inheritance(
+                ht, self._inheritance_mode, inheritance_filter, sorted_wes_family_sample_data,
+                sample_type='WES',
+            )
+            ht = self._annotate_families_inheritance(
+                ht, self._inheritance_mode, inheritance_filter, sorted_wgs_family_sample_data,
+                sample_type='WGS',
+            )
+            ht = self._post_process_inheritance_both_sample_types(ht)
+            ht = ht.filter(
+                ht.wes_passes.any(hl.is_defined) | ht.wgs_passes.any(hl.is_defined)
+            ).select_globals('family_guids', 'family_guids_1')
+
+        return ht, comp_het_ht
+
+    def _post_process_inheritance_both_sample_types(self, ht: hl.Table):
+        if self._inheritance_mode == DE_NOVO:
+            def _validate_de_novo(sample_type_1_passes, sample_type_2_passes):
+                # Remove a variant if it was filtered out due to inheritance in one sample type
+                # and is de novo in the other sample type but there is only one affected sample.
```

Comment:
Is there a reason we need to special case "de novo in the other sample type but there is only one affected sample" - shouldn't the behavior just be if it passes the filter in either return it in both, if it fails in both don't return it?

Reply:
This handles case 2 [here](https://docs.google.com/presentation/d/1hqDV8ulhviUcR5C4PtNUqkCLXKDsc6pccgFVlFmWUAU/edit#slide=id.p) - and the noted outcome is to not return either variant. This is the only case that violates the OR logic that the other cases follow (if it passes the filter in either return it in both) but it would be much simpler to not have to handle this explicitly and just return it in both here.

I didn't re-implement this when I moved the both sample types code into the mito query class, but I can. 

The case is unit tested in test_both_sample_types_search (family 5). Without this special logic, a de novo search will return the variant. With the logic, it will be filtered out.